DoLah/Parser/DateTimeParser.cpp

```diff
@@ -56,44 +58,88 @@ namespace DoLah {
         std::string out;
         for (size_t m = 0; m < monthPattern.size(); m++) {
             if (std::regex_match(str, std::regex(monthPattern.at(m), std::regex_constants::icase))) {
-                return (int) m;
+                return (int)m;
             }
         }
 
-        return NULL;
+        return REJECT;
     }
 
-    int DateTimeParser::getYear(std::string str) {
-        if (str.length() != 2 && str.length() != 4) {
-            return NULL;
-        } else if (!ParserLibrary::isDecimal(str)) {
-            return NULL;
+    int DateTimeParser::getYear(std::string year) {
+        if (year.length() != 2 && year.length() != 4) {
+            return REJECT;
+        } else if (!ParserLibrary::isDecimal(year)) {
+            return REJECT;
         }
 
-        return std::stoi(str);
+        if (year.length() == 2) {
+            year = CENTURY + year;
+        }
+
+        return std::stoi(year);
     }
 
+
+
     std::tm DateTimeParser::toDateFormat(std::vector<std::string> strArr) {
+        std::tm output;
+
+        std::vector<std::string> cleanArr = strArr;
+        if (cleanArr.size() == 1) {
+            std::string str = cleanArr.at(0);
+            for (size_t i = 0; i < dateDividers.size(); i++) {
+                cleanArr = ParserLibrary::explode(str, dateDividers.at(i));
+                if (cleanArr.size() > 1) {
+                    break;
+                }
+            }
+        }
+        cleanArr = ParserLibrary::removeElementsFromStringVector(cleanArr, decorators);
+
+        try {
+            output = classifyDate(cleanArr);
+        } catch (std::invalid_argument e) {
+            throw e;
+        }
+
+        return output;
+    }
+
+    std::tm DateTimeParser::classifyDate(std::vector<std::string> strArr) {
+        std::tm output;
+        try {
+            output = checkDMYformat(strArr);
+        } catch (std::invalid_argument e) {
+            try {
+                output = checkMDYformat(strArr);
+            } catch (std::invalid_argument e) {
+                throw e;
+            }
+        }
+        return output;
+    }
+
+    std::tm DateTimeParser::checkDMYformat(std::vector<std::string> strArr) {
         time_t t = time(0);
-        struct tm output;
+        std::tm output;
         localtime_s(&output, &t);
 
-        int day;
-        int month;
-        int year;
+        int day = -1;
+        int month = -1;
+        int year = -1;
 
         size_t size = strArr.size();
-        if ((day = getDay(strArr.at(0))) != NULL) {
+        if ((day = getDay(strArr.at(0))) != REJECT) {
```

Comment:
I don't understand what you mean. Why would Java throw exception if you assign NULL only in a separate line? (I think even in the same line or different line, whether it will throw exception or not, it is two different and unrelated things in both C++ and Java)